### PR TITLE
feat(smart-contracts): adding tron implementation to the library

### PR DESCRIPTION
**Pull request by fmeichtri-simtlix**
- Add Tron Structure
- Add Tron Contracts and Methods classes
- Add Tron Decoder
- Add Tron examples and tests
Associated ticket [STLX-6](https://bitgoinc.atlassian.net/browse/STLX-6)
**This PR is blocked by https://github.com/BitGo/eth-contracts/pull/20**

### CODE REVIEW

**Comment by fmeichtri-simtlix** [comment]
This PR depends on https://github.com/BitGo/eth-contracts/pull/20, once the https://github.com/BitGo/eth-contracts/pull/20 is merged we are going to rebase this branch and point the PR to master branch

File: src/base/contracts/contractInstances.ts [hunk 1]
```diff
@@ -16,6 +16,23 @@ export function listContractTypes(abiDirPath: string): string[] {
...
+ * @param contractName The name of the contract to read the ABI for
+ * @param accessAbiValues Access values to the ABI, some contracts has sub levels to access the ABI, e.g Tron has {"entrys" : [..ABI]}
+ * 
+ */
+export function getJsonAbi(contractName: string, abiDirPath : string, accessAbiValues : string[] = []) {
+  const jsonAbi = fs.readFileSync(path.join( __dirname, `${abiDirPath}/${contractName}.json`), 'utf-8');
+  ensure(isValidJSON(jsonAbi), `Invalid JSON: ${jsonAbi}`);
+  let parsedJsonAbi = JSON.parse(jsonAbi);
+  if (accessAbiValues) {
+    accessAbiValues.forEach(entry => {return parsedJsonAbi = parsedJsonAbi[entry];});
```
Comment (twtaylorbitgo) [inline src/base/contracts/contractInstances.ts:30]: what exactly is this trying to accomplish? it's weird we have an assignment operator in a return

Comment (twtaylorbitgo) [reply src/base/contracts/contractInstances.ts:30]: or rather think of it this way - is what this doing clear to a naive developer?

**Review by twtaylorbitgo — CHANGES_REQUESTED** [review]

File: src/base/contracts/contractInstances.ts (hunk 1 above)
Comment (fmeichtri-simtlix) [reply src/base/contracts/contractInstances.ts]: I changed this implementation, now should be more clear what this function do. Thanks!

**Review by twtaylorbitgo — APPROVED** [review]

**Pull request merged**